lib/rspec/support/ruby_features.rb

```diff
@@ -65,6 +66,29 @@ def supports_exception_cause?
         end
       end
 
+      UnsupportedVersionError = Class.new(StandardError)
+
+      begin
+        VersionChecker.new('Ruby', RUBY_VERSION, '1.9.0').check_version!
+
+        if Ruby.jruby?
+          # JRuby 1.7.5 - 1.7.13 actually supports Ripper but it's too slow.
+          # See benchmarks/ripper.rb.
+          VersionChecker.new('JRuby', JRUBY_VERSION, '1.7.14').check_version!
+
+          # Ripper on JRuby 9.0.0.0 reports wrong line number.
+          raise UnsupportedVersionError if JRUBY_VERSION.start_with?('9.')
+        end
+
+        def ripper_supported?
+          true
+        end
+      rescue LibraryVersionTooLowError, UnsupportedVersionError
```

Comment:
Exceptions for control flow is always one of those things that feels smelly.  Instead, WDYT of making `VersionChecker#compare_version` public and using that to find out how the version compares w/o needing to use exceptions?

Actually, it looks like nothing is using the version checker so you should feel free to change it's API to whatever you need :).